api/handlers.go

```diff
@@ -462,7 +462,11 @@ func getContainerJSON(c *context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	client, scheme := newClientAndScheme(c.tlsConfig)
+	client, scheme := container.Engine.HTTPClientAndScheme()
+	if client == nil {
+		httpError(w, "cannot connect to docker engine", http.StatusInternalServerError)
```

Comment:
Small detail - is this the same error message as returned by engine on failed connection?